sdk/canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/Identifier.java

```diff
@@ -14,12 +14,25 @@ public final class Identifier {
   private final String entityName;
 
   public Identifier(
-      @NonNull String packageId, @NonNull String moduleName, @NonNull String entityName) {
-    this.packageId = packageId;
+          @NonNull String packageId,
+          @NonNull String packageName,
+          @NonNull String moduleName,
+          @NonNull String entityName) {
+    // for backwards compatibility currently omitting
+    if (packageId.isEmpty()) {
```

Comment:
Is this the right form of backwards compatibility? 

- It seems to me the code generated would work with a ledger 3.2, I don't this is required. We can assume that the client code generated will only be used against 3.3 and above
- We want to generate code that will work with 3.3, but won't break against a 3.4 ledger, which means we do want to populate the packageName field